dist/prototypes.py

```diff
@@ -53,42 +70,96 @@ def extract_prototypes(filename, regexp):
     return ret
 
 # Build function prototypes from a list of files.
-def fn_prototypes(fns, tests, name):
+def fn_prototypes(public_fns, private_fns, tests, name):
     for sig in extract_prototypes(name, r'\n[A-Za-z_].*\n__wt_[^{]*'):
-        fns.append(sig)
+        public_fns.append(sig)
 
     for sig in extract_prototypes(name, r'\n[A-Za-z_].*\n__wti_[^{]*'):
-        fns.append(sig)
+        private_fns.append(sig)
 
     for sig in extract_prototypes(name, r'\n[A-Za-z_].*\n__ut_[^{]*'):
         tests.append(sig)
 
-# Write results and compare to the current file.
-# Unit-testing functions are exposed separately in their own section to
-# allow them to be ifdef'd out.
-def output(fns, tests, f):
-    tmp_file = '__tmp_prototypes' + str(os.getpid())
+# Create a new header file with function declarations.
+# As this is a new file also generate boilerplate like #pragma once
+def create_new_header(tmp_file, fns, tests, private_file=""):
     tfile = open(tmp_file, 'w')
+    tfile.write(COPYRIGHT)
     tfile.write("#pragma once\n\n")
+
+    if private_file:
+        tfile.write(f"#include \"{private_file}\"\n\n")
+
+    tfile.write("/* DO NOT EDIT: automatically built by prototypes.py: BEGIN */\n\n")
+
     for e in sorted(list(set(fns))):
         tfile.write(e)
 
     tfile.write('\n#ifdef HAVE_UNITTEST\n')
     for e in sorted(list(set(tests))):
         tfile.write(e)
-    tfile.write('\n#endif\n')
 
+    tfile.write('\n#endif\n')
+    tfile.write("\n\n/* DO NOT EDIT: automatically built by prototypes.py: END */\n")
     tfile.close()
+
+# Update an existing header file with function declarations.
+# As this file already exists only modify content between the DO NOT EDIT flags.
+def update_existing_header(tmp_file, fns, tests, f):
+    with open(f, 'r') as file:
+        lines = file.readlines()
+
+    start_line = lines.index('/* DO NOT EDIT: automatically built by prototypes.py: BEGIN */\n')
+    end_line = lines.index('/* DO NOT EDIT: automatically built by prototypes.py: END */\n')
+
+    # Safety check: We should always at least one function declaration in the file already
```

Comment:
Why is there an expectation of one function declaration?

Reply:
I don't recall the reason I added it originally.. I removed the assert in [2feddd8](https://github.com/wiredtiger/wiredtiger/pull/11064/commits/2feddd89cb4c65c31a34b4ebacb9b6c8473324e5) and confirmed that it's not an issue if the DO_NOT_EDIT lines are directly next to each other